### PR TITLE
add timelock queries

### PR DESCRIPTION
**Pull request by huangsuyu**

### CODE REVIEW

File: common/types/timelock.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+func (a TimeLockRecords) Swap(i, j int) {
+	a[i], a[j] = a[j], a[i]
+}
+func (a TimeLockRecords) Less(i, j int) bool {
+	return a[i].Id < a[j].Id
+}
+
+
+// Params for query 'custom/gov/proposal'
```
Comment (yutianwu) [inline common/types/timelock.go]: comment is not right

File: client/rpc/dex_client.go [hunk 2]
```diff
@@ -5,12 +5,17 @@ import (
...
 	"github.com/binance-chain/go-sdk/common/types"
 	"github.com/binance-chain/go-sdk/types/tx"
+
 )
 
 const (
 	AccountStoreName = "acc"
 	TokenStoreName   = "tokens"
 	ParamABCIPrefix  = "param"
+	MsgRoute = "timelock"
```
Comment (yutianwu) [inline client/rpc/dex_client.go]: it would be better to format these lines

File: e2e/e2e_rpc_test.go [hunk 3]
```diff
@@ -20,11 +20,11 @@ import (
 )
 
 var (
-	nodeAddr           = "tcp://127.0.0.1:80"
+	nodeAddr           = "tcp://data-seed-pre-0-s1.binance.org:80"
```
Comment (yutianwu) [inline e2e/e2e_rpc_test.go]: shall we change this

File: common/types/timelock.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import "time"
+
+type TimeLockRecord struct {
+	Id          int64     `json:"id"`
+	Description string    `json:"description"`
+	Amount      Coins `json:"amount"`
+	LockTime    time.Time `json:"lock_time"`
+}
+
+type TimeLockRecords []TimeLockRecord
```
Comment (yutianwu) [inline common/types/timelock.go]: you do not need to copy these

File: common/types/timelock.go (hunk 1 above)
Comment (huangsuyu) [reply common/types/timelock.go]: changed!

File: client/rpc/dex_client.go [hunk 5]
```diff
@@ -5,12 +5,17 @@ import (
 	"fmt"
 	"github.com/binance-chain/go-sdk/common/types"
 	"github.com/binance-chain/go-sdk/types/tx"
+
```
Comment (yutianwu) [inline client/rpc/dex_client.go]: delete this empty line

File: client/rpc/dex_client.go [hunk 6]
```diff
@@ -5,12 +5,17 @@ import (
...
 )
 
 const (
 	AccountStoreName = "acc"
```
Comment (guagualvcha) [inline client/rpc/dex_client.go]: could you do `gofmt -w -s xx.go` to these modified files.

File: client/rpc/dex_client.go [hunk 7]
```diff
@@ -5,12 +5,17 @@ import (
...
 	TokenStoreName   = "tokens"
 	ParamABCIPrefix  = "param"
+	MsgRoute         = "timelock"
```
Comment (guagualvcha) [inline client/rpc/dex_client.go]: `MsgRoute ` is too common, please change to a name that relate to `timelock`

File: client/rpc/dex_client.go [hunk 8]
```diff
@@ -240,6 +245,74 @@ func (c *HTTP) GetDepth(tradePair string) (*types.OrderBook, error) {
 	return &ob, nil
 }
 
+func (c *HTTP) GetTimelocks(address string) (types.TimeLockRecords, error) {
+
+	hexaddr, err := types.AccAddressFromBech32(address)
```
Comment (guagualvcha) [inline client/rpc/dex_client.go]: `hexaddr ` to `addr`

File: client/rpc/dex_client.go [hunk 9]
```diff
@@ -240,6 +245,74 @@ func (c *HTTP) GetDepth(tradePair string) (*types.OrderBook, error) {
 	return &ob, nil
 }
 
+func (c *HTTP) GetTimelocks(address string) (types.TimeLockRecords, error) {
```
Comment (guagualvcha) [inline client/rpc/dex_client.go]: The response is slice, but definition here is just `types.TimeLockRecords`,  I think compile will fail.

File: common/types/timelock.go [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
```
Comment (guagualvcha) [inline common/types/timelock.go]: No need to define `TimeLockRecords `

File: client/rpc/dex_client.go (hunk 6 above)
Comment (huangsuyu) [reply client/rpc/dex_client.go]: solved!

File: client/rpc/dex_client.go (hunk 8 above)
Comment (huangsuyu) [reply client/rpc/dex_client.go]: solved!

File: common/types/timelock.go (hunk 10 above)
Comment (huangsuyu) [reply common/types/timelock.go]: solved!

**Comment by guagualvcha** [comment]
Can you add these two need method into the interface `type DexClient interface `

File: client/rpc/dex_client.go [hunk 11]
```diff
@@ -11,6 +11,9 @@ const (
 	AccountStoreName = "acc"
 	TokenStoreName   = "tokens"
 	ParamABCIPrefix  = "param"
+	TimeLockMsgRoute = "timelock"
+	QueryTimeLocks   = "timelocks"
```
Comment (guagualvcha) [inline client/rpc/dex_client.go]: To keep the code style consistent,  `QueryTimeLocks ` and `QueryTimeLock ` please just use it, don't define the constant.

File: client/rpc/dex_client.go [hunk 12]
```diff
@@ -240,6 +243,73 @@ func (c *HTTP) GetDepth(tradePair string) (*types.OrderBook, error) {
...
+	}
+
+	params := types.QueryTimeLocksParams{
+		Account: addr,
+	}
+
+	bz, err := c.cdc.MarshalJSON(params)
+
+	if err != nil {
+		return nil, fmt.Errorf("incorrectly formatted request data %s", err.Error())
```
Comment (guagualvcha) [inline client/rpc/dex_client.go]: ` fmt.Errorf("marshal params failed  %v", err)`

File: client/rpc/dex_client.go [hunk 13]
```diff
@@ -240,6 +243,73 @@ func (c *HTTP) GetDepth(tradePair string) (*types.OrderBook, error) {
...
+	}
+
+	rawRecords, err := c.ABCIQuery(fmt.Sprintf("custom/%s/%s", TimeLockMsgRoute, QueryTimeLocks), bz)
+
+	if err != nil {
+		return nil, fmt.Errorf("error query %s", err.Error())
```
Comment (guagualvcha) [inline client/rpc/dex_client.go]: just `return nil, err`

File: client/rpc/dex_client.go (hunk 12 above)
Comment (huangsuyu) [reply client/rpc/dex_client.go]: edited!

File: client/rpc/dex_client.go (hunk 13 above)
Comment (huangsuyu) [reply client/rpc/dex_client.go]: edited!

**Pull request merged**